pkg/backingstore/reconciler.go

```diff
@@ -826,14 +832,22 @@ func (r *Reconciler) fixAlternateKeysNames() {
 // ReconcileExternalConnection handles the external connection using noobaa api
 func (r *Reconciler) ReconcileExternalConnection() error {
 
-	// TODO we only support creation here, but not updates
 	if r.ExternalConnectionInfo != nil {
 		return nil
 	}
 	if r.AddExternalConnectionParams == nil {
 		return nil
 	}
 
+	if r.UpdateExternalConnectionParams != nil {
+		err := r.NBClient.UpdateExternalConnectionAPI(*r.UpdateExternalConnectionParams)
```

Comment:
what is the reason for not calling checkExternalConnection before updating? I know it's called from inside updateExternalConnection in noobaa-core but I think it's needed for keeping the same behaviour as for AddExternalConnection

Reply:
calling checkExternalConnection before updating will return a `CONNECTION_ALREADY_EXIST` error, as you said it's called from inside updateExternalConnection, and if there is something wrong with the connection it will be checked there.

Reply:
it will be checked but you won't throw a persistent error and the whole logic there is missing.